src/environment_windows.go

```diff
@@ -105,31 +105,57 @@ func (env *environment) getCachePath() string {
 }
 
 //
-// Takes a registry path like "HKLM\Software\Microsoft\Windows NT\CurrentVersion" and a key under that path like "CurrentVersion" (or "" if the (Default) key is required).
-// Returns a bool and string:
+// Takes a registry path to a key like
+//		"HKLM\Software\Microsoft\Windows NT\CurrentVersion\EditionID"
 //
-//   true and the retrieved value formatted into a string if successful.
-//   false and the string will be the error
+// The last part of the path is the key to retrieve.
 //
-func (env *environment) getWindowsRegistryKeyValue(regPath, regKey string) (string, error) {
-	env.trace(time.Now(), "getWindowsRegistryKeyValue", regPath, regKey)
-
-	// Extract root HK value and turn it into a windows.Handle to open the key.
-	regPathParts := strings.SplitN(regPath, "\\", 2)
+// If the path ends in "\", the "(Default)" key in that path is retrieved.
+//
+// Returns a variant type if successful; nil and an error if not.
+//
+func (env *environment) getWindowsRegistryKeyValue(path string) (*windowsRegistryValue, error) {
+	env.trace(time.Now(), "getWindowsRegistryKeyValue", path)
+
+	// Format:
+	// "HKLM\Software\Microsoft\Windows NT\CurrentVersion\EditionID"
+	//   1  |                  2                         |   3
+	//
+	// Split into:
+	//
+	// 1. Root key - extract the root HKEY string and turn this into a handle to get started
+	// 2. Path - open this path
+	// 3. Key - get this key value
+	//
+	// If 3 is "" (i.e. the path ends with "\"), then get (Default) key.
+	//
+	regPathParts := strings.SplitN(path, "\\", 2)
 
 	regRootHKeyHandle := getHKEYHandleFromAbbrString(regPathParts[0])
 	if regRootHKeyHandle == 0 {
 		errorLogMsg := fmt.Sprintf("Error, Supplied root HKEY value not valid: '%s'", regPathParts[0])
 		env.log(Error, "getWindowsRegistryKeyValue", errorLogMsg)
-		return "", errors.New(errorLogMsg)
+		return nil, errors.New(errorLogMsg)
 	}
 
+	// Strip key off the end.
+	lastSlash := strings.LastIndex(regPathParts[1], "\\")
```

Comment:
same here, we expect as second element, I would validate if we have all entries before continuing.

Reply:
yup - validation needed.